File: backend/src/main/java/com/horain/repository/ActivityTypeRepository.java
package com.horain.repository;

import com.horain.model.ActivityType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

/**
 * JPA repository for activity types (natures with TJM).
 */
public interface ActivityTypeRepository extends JpaRepository<ActivityType, String> {

    List<ActivityType> findAllByOrderByCodeAsc();
}

File: backend/src/main/kotlin/com/horain/repository/ActivityTypeRepository.kt
package com.horain.repository

import com.horain.model.ActivityType
import org.springframework.data.jpa.repository.JpaRepository

/**
 * JPA repository for activity types (natures with TJM).
 */
interface ActivityTypeRepository : JpaRepository<ActivityType, String> {

    fun findAllByOrderByCodeAsc(): List<ActivityType>
}
